src/checkout-buttons/strategies/googlepay/googlepay-braintree-button-strategy.ts

```diff
@@ -0,0 +1,118 @@
+import { GooglePayPaymentProcessor } from '../../../payment/strategies/googlepay';
+import { CheckoutButtonInitializeOptions } from '../../checkout-button-options';
+import CheckoutButtonStrategy from '../checkout-button-strategy';
+
+export default class GooglePayBraintreeButtonStrategy extends CheckoutButtonStrategy {
+    private _methodId?: string;
+    private _checkout?: Checkout;
+    private _walletButton?: HTMLElement;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _formPoster: FormPoster,
+        private _checkoutActionCreator: CheckoutActionCreator,
+        private _googlePayPaymentProcessor: GooglePayPaymentProcessor
+    ) {
+        super();
+    }
+
+    initialize(options: CheckoutButtonInitializeOptions): Promise<void> {
+        const { containerId, methodId } = options;
+
+        if (!containerId || !methodId) {
+            throw new InvalidArgumentError('Unable to proceed because "containerId" argument is not provided.');
+        }
+
+        if (this._isInitialized[containerId]) {
+            return super.initialize(options);
+        }
+
+        this._methodId = methodId;
+
+        return this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout())
+            .then(stateCheckout => {
+                this._checkout = stateCheckout.checkout.getCheckout();
+                if (!this._checkout || !this._checkout.cart.id) {
+                    throw new MissingDataError(MissingDataErrorType.MissingCart);
+                }
+
+                return this._googlePayPaymentProcessor.initialize(this._getMethodId())
+                    .then(() => {
+                        this._walletButton = this._createSignInButton(containerId);
+
+                        if (this._walletButton) {
+                            this._walletButton.addEventListener('click', this._handleWalletButtonClick);
+                        }
+                    });
+            }).then(() => super.initialize(options));
+    }
+
+    deinitialize(): Promise<void> {
+        if (!this._isInitialized) {
+            return super.deinitialize();
+        }
+
+        if (this._walletButton && this._walletButton.parentNode) {
+            this._walletButton.parentNode.removeChild(this._walletButton);
+            this._walletButton.removeEventListener('click', this._handleWalletButtonClick);
+            this._walletButton = undefined;
+        }
+
+        return this._googlePayPaymentProcessor.deinitialize()
+            .then(() => super.deinitialize());
+    }
+
+    private _createSignInButton(containerId: string): HTMLElement {
+        const container = document.querySelector(`#${containerId}`);
+
+        if (!container) {
+            throw new InvalidArgumentError('Unable to create sign-in button without valid container ID.');
+        }
+
+        const googlePayButton = this._googlePayPaymentProcessor.createButton(() => this._onPaymentSelectComplete);
```

Comment:
You're getting a lint error because you're not defining the interface correctly. This is what you have currently

```ts
createButton(onClick: () => {}, buttonType: ButtonType, buttonColor: ButtonColor): HTMLElement
```

In other words, `onClick` is defined as a function that could return an empty object (`{}`). When you call, `createButton(() => this._onPaymentSelectComplete)`, you're in fact passing in a function that returns an empty object. That's why the compiler doesn't complain. But that's not what you really want. You actually want to call `_onPaymentSelectComplete` when the button is clicked. But calling it will trigger a compiler error because `_onPaymentSelectComplete` returns `void` instead. So what you want instead is to update the interface of `createButton` to better reflect your intention.

```ts
createButton(onClick: () => void, buttonType: ButtonType, buttonColor: ButtonColor): HTMLElement
```

Reply:
@metalvegetarianoprogresivo @vmparra I'm not seeing this comment as being resolved in the code. Can you please address?